File: app_project/backend/app/schemas/news/donor.py
from pydantic import BaseModel
from datetime import datetime

class DonorBase(BaseModel):
    full_name: str
    donation_amount: float
    message: str | None = None

class DonorCreate(DonorBase):
    image: bytes | None = None

class DonorUpdate(DonorBase):
    image: bytes | None = None

class DonorResponse(DonorBase):
    id: int
    donated_at: datetime
    image: bytes | None = None

    class Config:
        from_attributes = True

File: app_project/frontend/src/models/news/donor.ts
// models/news/donor.ts

export interface DonorBase {
  full_name: string;
  donation_amount: number;
  message?: string | null;
}

export interface DonorCreate extends DonorBase {
  image?: File | null;
}

export interface DonorUpdate extends DonorBase {
  image?: File | null;
}

export interface DonorResponse extends DonorBase {
  id: number;
  donated_at: string; // ISO date string
  image?: string | null; // base64 or URL if handled separately
}
